storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -1949,6 +1934,16 @@ bool copy_incremental_over_full() {
           goto cleanup;
         }
       }
+      /* remove archive dir if exists */
+      snprintf(path, sizeof(path), "%s/%s", ROCKSDB_SUBDIR, "archive");
+      if (directory_exists(path, false)) {
+        if (rmdir(path)) {
+          msg("xtrabackup: error: unable to remove directory '" ROCKSDB_SUBDIR
+              "'\n");
+          ret = false;
+          goto cleanup;
+        }
+      }
```

Comment:
Are we copying all archive dirs (including archive log files in it) to backup dir and later removing them from backup dir?

Reply:
We back it up on --backup and remove it if we are doing --prepare of an incremental. We remove it at copy_incremental_over_full - MyRocks is like MyISAM, we always do a full copy.

Reply:
Few questions: 1. are these WAL files deleted by server on TTL? what if xtrabackup sees them, tries to copy but before they are copied, server deletes them
2. Can we skip entirely copying them?

Reply:
1 - We disable rocksdb file deletions via https://www.percona.com/doc/percona-server/8.0/myrocks/variables.html#rocksdb_disable_file_deletions
2 - yes, seems like skip copying the archive folder is an option too (makes more sense IMHO) will code it.

Reply:
@satya-bodapati fixed - Sorry, had to rebase after #1155 and did a force-push.

Reply:
```
marcelo@marce-bld:/work/pxb/ins/8.0/xtrabackup-test/test$ ./run.sh -r /dev/shm/ -s rocksdb
Using /dev/shm//var as test root
Detecting server version...
Running against Percona Server 8.0.25-15-debug (XtraDB 8.0.25-15)
Using 'xtrabackup' as xtrabackup binary
Autodetected number of cores: 32
Limiting to 16 to avoid excessive resource consumption
Using 16 parallel workers
Running suite(s): rocksdb

==============================================================================
TEST                                   WORKER    RESULT     TIME(s) or COMMENT
------------------------------------------------------------------------------
ddl                                      w5	[passed]    27
wal_dir_rel                              w13	[passed]    29
wal_dir_abs                              w12	[passed]    31
datadir                                  w4	[passed]    35
wal_dir_abs_archive                      w11	[passed]    39
compress                                 w2	[passed]    48
local_max_age                            w6	[passed]    48
local                                    w7	[passed]    48
stream_compress                          w9	[passed]    47
compress_encrypt                         w1	[passed]    49
stream_compress_encrypt                  w8	[passed]    48
stream                                   w10	[passed]    48
cross_engine_trx                         w3	[passed]    54
==============================================================================
Spent 551 of 68 seconds executing testcases

SUMMARY: 13 run, 13 successful, 0 skipped, 0 failed

See results/ for detailed output
```